benchmarks/triangulated_surface_distance/generator.hpp

```diff
@@ -0,0 +1,269 @@
+/****************************************************************************
+ * Copyright (c) 2024 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_HyperTriangle.hpp>
+
+#include <Kokkos_MathematicalConstants.hpp>
+
+#if KOKKOS_VERSION >= 40400
+#include <Kokkos_Array.hpp>
+using Array = Kokkos::Array;
+#else
+template <class T, size_t N>
+struct Array
+{
+  static constexpr auto size() { return N; }
+  KOKKOS_FUNCTION constexpr T &operator[](int i) { return _data[i]; }
+  KOKKOS_FUNCTION constexpr T const &operator[](int i) const
+  {
+    return _data[i];
+  }
+
+  T _data[N];
+};
+#endif
+
+static auto icosahedron()
```

Comment:
Why static?  Were you getting an error about multiple definition?  Any particular reason to prefer static over inline?

Reply:
Yeah, I thought to be careful against multiple definition. Was not getting an error. Inline could work, though the function is quite large for that. I'm just going to drop it.